### DISCUSSION TITLE
Achievement moderators, commit verification, and contribution credit

### DISCUSSION THREAD

**Discussion by julkasalsa** [RetroAchievements/RAWeb#685] [Ideas]
First of all the description under the achievement could be renamed to "Started by" instead of "Created by" (and the related field in the achievement editor could be renamed to "Intiator" instead of "Author"). 
From now on every commit of the modification of the code would need to be verified by the site staff, before the update would be allowed. 
These verifications could be handled by moderators (new rank for more trusty developers), and each commit and verification would visible on the site log, allowed to check by all developers and up.
After the commit being verified, the developer who made it would be credited as contributor.
It would be necessary to keep the copy (or copies) of the previous version of the code, to bring the code back to the last working version.

**Comment by kdecks** [comment]
I agree with this concept as a whole, but have notes and additions.

---
>renamed to "Started by" instead of "Created by" 

There's no need for this change. As the primary meaning of *created* means, "To cause to exist; bring into being. It's synonymous with *started*.
### Here's a way I could see it presented to players:
![image](https://user-images.githubusercontent.com/32706333/36343478-11e48724-13ca-11e8-988f-0caa0eecdfc7.png)
And if they click the link it will expand showing all the changes, by who and when.

---
>These verifications could be handled by moderators (new rank for more trusty developers)

This concept should go into its own issue. What would moderators do that devs and admin wouldn't? (including this)

---
> verified by the site staff [moderators]


Even with moderators this feels like unnecessary busy work. I'll lay out below how we can get the most out of the concept and still be low maintenance.

---
> each commit ... would visible on the site log

For reference and linking (#11)

---
> It would be necessary to keep the copy (or copies) of the previous version of the code

### This is Dev tools right now on the achievement page:
![image](https://user-images.githubusercontent.com/32706333/36343597-5c5f4b8a-13cb-11e8-80a2-7b80bb5a578c.png)

### Here's how it could work on **Achievement pages ⇒ Dev tools**.

A version history of every update would be presented including the author of the change and when.
- These versions each would be expandable so you could see what changed when and compare versions.
- You would be able to change the version easily from the dev tools here.
- When a developer/admin reverts to an older version this change this change is added to update history, put on the bottom. This way you can make changes to achievements at any state of it's published history, no work could be lost, and you could easily revert unnecessary changes.

- [ ] I'd like to make a visual mock-up of this idea soon.
---
As for how it would be accomplished that's another thing entirely. It's a big overhaul no doubt, but would prevent and fix many problem in the long run.

**Comment by julkasalsa** [comment]
> There's no need for this change. As the primary meaning of created means, "To cause to exist; bring into being. It's synonymous with started.

"Started" is more specific in this matter, because it's about developer who brought the first version of the code, which not always will be action that is more momentous in front of all other contributions. Created is more strong by merit in my opinion.

**Comment by Jamiras** [comment]
Most applications that have change tracking have four standard audit fields: "Created", "Created By", "Updated" and "Updated By". "Created" and "Updated" are timestamps. "Created By" and "Updated By" are user IDs. The primary record has the current values, and those four fields. An associated audit log table has entries for each change to the record. If historical data must be maintained, the previous value for each change would be included in the associated audit log entry.
